File: Research/JO_matlab codes/Aed.m
%%Code to calculate A. Make sure to rename the created file every time.
%%Also cancel all the auxiliary variables created so far but q.
clear all;
e=0; q =0; RI = 0; O = 0; Sed = 0;
SumAed = 0; LifeTime = 0; BrRatio = 0;
prompt = 'How many lower transitions?: '; 
q = input(prompt);

promt = 'What is the Refractive Index?: ';
RI = input(promt);

for n = 1:1:3
  prompt = 'Enter the value of JO parameters in order and press enter after each entry: ';
  O(1,n) = input(prompt); 
 end
for n=1:1:q
 prompt = 'J value at excited state ';
    e(n,1)=input(prompt);
   prompt = 'Peak wavelength of emission ';
    e(n,2)=input(prompt);
   prompt = 'Enter U2: ';
    e(n,3)=input(prompt);
   prompt = 'Enter U4: '; 
    e(n,4)=input(prompt);
   prompt = 'Enter U6: '; 
    e(n,5)=input(prompt);
   Sed = e(n,3)*O(1,1)+e(n,4)*O(1,2)+e(n,5)*O(1,3)    
    e(n,6)= 7.23E10*RI*((RI^2+2)/3)^2*Sed/((2*e(n,1)+1)*(e(n,2)*1E-9)^3);
    SumAed = SumAed + e(n,6);
end
% calculation of Life time of states
LifeTime = 1 / SumAed;

% calculation of Branching ratio
for n = 1:1:q
BrRatio(1,n) = e(n,6) / SumAed;
end

%e(n,6) is the calculated Aed ;
%Check if a Amd occurs for the given transition.

File: Research/Python/Aed.py
#Code to calculate A. Make sure to rename the created file every time.
#Also cancel all the auxiliary variables created so far but q.

import numpy as np
q = int(input('How many lower transitions?:'))
RI = float(input('What is the Refractive Index?:'))

#def A(q, RI):
O = np.empty(3, float)
print('Enter the value of JO parameters in order and press enter after each entry:')
for i in range(0, 3):
    O[0][i] = float(input())
e = []
for e1 in range(0, q):
    e.append(np.empty(6, float))
np.array(e)
SumAed = 0
for n in range(0, q):
    e[n][0] = float(input("J value at excited state:"))
    e[n][1] = float(input("Peak wavelength of emission:"))
    e[n][2] = float(input("Enter U2:"))
    e[n][3] = float(input("Enter U4:"))
    e[n][4] = float(input("Enter U6:"))
    Sed = e[n][2] * O[0][0] + e[n][3] * O[0][1] + e[n][4] * O[0][2]
    e[n][5] = 7.23e10 * RI * ((RI ** 2 + 2) / 3) ** 2 * Sed / ((2 * e[n][0] + 1) * (e[n][1] * 1e-9) ** 3)
    SumAed += e[n][5]

#calculation of Life time of states
LifeTime = 1 / SumAed

#calculation of Branching ratio
BrRatio = np.array([])
BrRatio1 = np.array([])
for br in range(0, q):
    BrRatio1.append(0)
BrRatio.append(BrRatio1)
for i in range(0, q):   
    BrRatio[0][n] = e[n][5] / SumAed

#e(n,6) is the calculated Aed ;
#Check if a Amd occurs for the given transition.
